tests/testsuite/cargo_remove/gc_patch/stderr.log

```diff
@@ -1,3 +1,8 @@
     Removing bar from dependencies
-    Updating git repository `[ROOTURL]/bar2`
+    Updating git repository `[ROOTURL]/bar[..]`
+    Updating git repository `[ROOTURL]/bar[..]`
     Updating `dummy-registry` index
+warning: Patch `bar v0.1.0 ([..])` was not used in the crate graph.
+Perhaps you misspelled the source URL being patched.
+Possible URLs for `[patch.<URL>]`:
+    [ROOT]/bar2
```

Comment:
This is a result of resolving twice. The first time unused patches are found so this is emitted. Later they are gc and re-resolve so we don't get the same warning twice

Reply:
This will be very confusing to users and should be resolved.

A quick hack would be to capture the verbosity, set it to quiet, resolve, then restore the verbosity.  We *might* be able to improve it with #12235 by pushing an "allow" on that warning and then popping it.

Reply:
Agreed. Is there any example of this in the code? I'm not quite sure how to do that

Reply:
Not that I can think of.  The relevant calls are
- https://doc.rust-lang.org/nightly/nightly-rustc/cargo/util/config/struct.Config.html#method.shell
- https://doc.rust-lang.org/nightly/nightly-rustc/cargo/core/shell/struct.Shell.html#method.set_verbosity
- https://doc.rust-lang.org/nightly/nightly-rustc/cargo/core/shell/struct.Shell.html#method.verbosity

Reply:
Nevermind I figured it out

Reply:
unfortunately, this is removing other status messages like 
```
 Updating `dummy-registry` index
```

Reply:
Could you post it so I can see?

Reply:
done

Reply:
Thank you for the guidance and the code snippet. The snippet helped me realize the bug here 👍